File: backend/internal/models/registration.go
package models

type Registration struct {
	DbModel
	UserID    uint
	User      User       `json:"user"`
	Questions []Question `type:"questions"`
}

type Question struct {
	DbModel
	RegistrationID uint
	Text           string `json:"text"`
	Answer         string `json:"answer"`
	Render         string `json:"render"`
}

File: frontend/models/Registration.ts
import type { User } from './User'

export interface Registration {
  // dbModel: DbModel
  // user_id: UserId
  user: User
  questions: Array<Question>
}

export interface Question {
  // dbModel: DbModel
  // registration_id: RegistrationId
  text: string
  answer: string
  render: string
}
